pkg/raft/raft.go

```diff
@@ -1184,7 +1184,7 @@ func (r *raft) tickElection() {
 		r.electionElapsed++
 	}
 
-	if r.promotable() && r.pastElectionTimeout() {
```

Comment:
From you second commit description: 
> If a replica is not promotable and we are past the randomizedElectionTimeout, we will keep try to campaign on every tick. 

I’m not following why we do that. The `promotable()` check ensures 3 things:
1. we are part of the active configuration. If not we don’t have a chance to win - nodoby will vote for us
2. we are not a learner. If we are, we aren’t a voter and shouldn’t campaign
3. we aren’t processing an incoming snapshot. I’m less clear on why we can’t campaign if we are

At least in cases 1 and 2, why would we proceed to campaign here?

Are we relying on the fact that `hup` method will check `promotable()` down the stack and not campaign anyway?